src/lib/utils/generate-salt.js

```diff
@@ -1,15 +1,15 @@
 /**
+ * @TODO Where does BaseElement come from?
+ *
  * Generates a random string that can be used to ensure uniqueness of
  * the elements id on a page.
  * @param {string} idPrefix An id prefix to be followed by the generated salt.
  *     Used to check the uniqueness of the outcome id (prefix + salt).
  * @return {string} Id salt.
  */
 export const generateIdSalt = (idPrefix) => {
-  const salt = Math.random()
-    .toString(36)
-    .substr(2, 9);
+  const salt = Math.random().toString(36).substr(2, 9);
   return document.getElementById(idPrefix + salt)
-    ? BaseElement.generateIdSalt(idPrefix)
+    ? BaseElement.generateIdSalt(idPrefix) // eslint-disable-line
```

Comment:
@robdodson @samthor Where does BaseElement come from?

Reply:
src/lib/components/BaseElement but it looks like they didn't import it here. Perhaps we're getting lucky that it's already in the bundle so this isn't throwing an error.

Reply:
Hrm, ok, but it doesn't seem like `generateIdSalt` is a method of BaseElement or LitElement.

Reply:
oh i see what you're saying.

yeah I don't know what's up with that... @devnook did you work on the generateIdSalt code?

Reply:
Nope, but here's the relevant commit: https://github.com/GoogleChrome/web.dev/pull/2229/commits/f714e3ea2b6bc9f1e0ac1117640d13f690d8a96a

Reply:
Thank you @devnook I'm modifying that now